addons/hr_payroll/report/payslip_report.py

```diff
@@ -0,0 +1,34 @@
+#-*- coding:utf-8 -*-
+
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class PayslipReport(models.AbstractModel):
+    _name = 'report.hr_payroll.report_payslip'
+
+    def get_payslip_lines(self, obj):
+        PayslipLine = self.env['hr.payslip.line']
+        res = []
+        ids = []
+        for id in range(len(obj)):
+            if obj[id].appears_on_payslip is True:
+                ids.append(obj[id].id)
```

Comment:
i think function can be improved and reduced only in one line. as it's find appears_on_payslip line and return ids of that cannot we directly use filtered.
now obj is recordset. first change variable name obj to payslip_lines
then 
return payslip.line_ids.filtered(lambda x: x.appears_on_payslip)